technician_client.py

```
@@ -0,0 +1,63 @@
+import socket
+import struct
+from typing import Any, Optional, Union
+
+import typer
+
+EXIT_COMMAND = "exit"
+
+
+def recv_struct(client: socket.socket, format: Union[str, bytes]) -> Any:
+    calculated_size = struct.calcsize(format)
+    received = client.recv(calculated_size)
+    if len(received) != calculated_size:
+        return None
+    return struct.unpack(format, received)[0]
+
+
+def recv_message(client: socket.socket) -> Optional[bytes]:
+    """Receives a message from the client, first receiving the message's length."""
+    length = recv_struct(client, "<L")
+    if length is None:
+        return None
+    message = client.recv(length)
+    if len(message) != length:
+        return None
+    return message
+
+
+def send_message(client: socket.socket, message: bytes) -> None:
+    """Sends the message to the client, with the length doubleword included before it."""
+    client.send(struct.pack("<L", len(message)) + message)
+
+
+def send_recv_command(client: socket.socket, command: str) -> bool:
```

Might be better to throw an exception when this function fails, also print debug information to know if the client disconnected due to an error. Currently the program just stops without saying a thing